okhttp/src/main/java/io/grpc/okhttp/OkHttpChannelProvider.java

```diff
@@ -33,18 +33,29 @@
 
 import io.grpc.Internal;
 import io.grpc.ManagedChannelProvider;
+import io.grpc.internal.GrpcUtil;
 
-/** Provider for {@link OkHttpChannelBuilder} instances. */
+/**
+ * Provider for {@link OkHttpChannelBuilder} instances.
+ */
 @Internal
 public final class OkHttpChannelProvider extends ManagedChannelProvider {
+
+  private final int priority;
+
+  public OkHttpChannelProvider() {
+    // Netty cannot run under the current AppEngine runtime sandbox
+    priority = (GrpcUtil.IS_RESTRICTED_APPENGINE || isAndroid()) ? 8 : 3;
```

Comment:
Can this be done in the `priority()` method like it used to be? We really want as little as possible happening during construction because any exceptions will cause trouble. `priority()` is also called rarely so performance of the method doesn't matter much.

If you really like it here, then using a static field seems more appropriate.